tests/e2e/IExecDataProtector/grantAccess.test.ts

```diff
@@ -128,17 +128,19 @@ describe('dataProtector.grantAccess()', () => {
   });
   it('fails if the app is not deployed', async () => {
     await expect(dataProtector.grantAccess({ ...input })).rejects.toThrow(
-      Error(
-        `Failed to grant access: No app found for id ${input.authorizedApp} on chain 134`
+      new WorkflowError(
+        'Failed to detect the app TEE framework',
+        Error(`No app found for id ${input.authorizedApp} on chain 134`)
       )
     );
   });
   it('fails if the app is not a TEE app', async () => {
     await expect(
       dataProtector.grantAccess({ ...input, authorizedApp: nonTeeAppAddress })
     ).rejects.toThrow(
-      Error(
-        'Failed to grant access: App does not use a supported TEE framework'
+      new WorkflowError(
+        'Failed to detect the app TEE framework',
+        Error('App does not use a supported TEE framework')
       )
     );
   });
```

Comment:
We can create a ticket in the next sprint to add unit tests, mocking, and testing all error-catching scenarios.

Reply:
yes I drafted a unit test implementation that works but I'm not completely happy with it in the way things are mocked or stubbed. you can have a look on it https://github.com/iExecBlockchainComputing/dataprotector-sdk/pull/68